### ISSUE TITLE
Extra whitespace added when block elements are separated by newlines

### ISSUE THREAD

**Issue by craiga** [matthewwithanm/python-markdownify#17]
When the input HTML contains newlines between block elements, extra whitespace is being injected.

For example:

```
>>> import markdownify
>>> markdownify.markdownify("<h1>First</h1>\n<h2>Second</h2>\n<h3>Third</h3>", heading_style=markdownify.ATX)
'# First\n\n ## Second\n\n ### Third\n\n'
```

Note that I'm only using ATX-style headers here to help demonstrate the problem.

**Comment by puhoy** [comment]
this is a problem when anything but a closing tag comes before the headline.
```
>>> import markdownify
>>> markdownify.markdownify(' <h1>First</h1>', heading_style=markdownify.ATX)
' # First\n\n'
>>> markdownify.markdownify('X<h1>First</h1>', heading_style=markdownify.ATX)
'X# First\n\n'
```
a fix would be to just prefix all headlines with a linebreak, which is not the nicest solution and breaks some tests. (but, i guess, would be a valid solution for most usecases?)

**Comment by adamb70** [comment]
I'm running into the same issue, any further thoughts to address this?

```python
>>> markdownify('<p>word</p> <h3>header</h>')
'word\n\n ### header\n\n'
```

The whitespace before `###` is enough to stop some renderers displaying the header.

I'm willing to make a PR with Puhoy's suggestion and address the tests if that is a desirable solution.

**Comment by desmap** [comment]
Running into the same, even with normal paragraphs.

**Comment by adamb70** [comment]
Not a great permanent solution, but I've just been monkey patching it for now.

```python
import markdownify


def patched_convert_hn(self, n, el, text, convert_as_inline):
    if convert_as_inline:
        return text

    style = self.options['heading_style']
    text = text.rstrip()
    if style == markdownify.UNDERLINED and n <= 2:
        line = '=' if n == 1 else '-'
        return self.underline(text, line)
    hashes = '#' * n
    if style == markdownify.ATX_CLOSED:
        return '\n%s %s %s\n\n' % (hashes, text, hashes)  # patched with extra \n
    return '\n%s %s\n\n' % (hashes, text)  # patched with extra \n

markdownify.MarkdownConverter.convert_hn = patched_convert_hn


patched_markdownify = markdownify
```

Then in your other files

```python
from .monkey_patch import patched_markdownify as markdownify

# Use as normal
markdownify.markdownify('X<h1>First</h1>')
...
```

**Comment by AlexVonB** [comment]
Hey all,

sorry for the late response. I joined the repo later and did not receive notifications for this issue.

The issue seems to be that we replace all new lines and tabs with a single space, leading to the space being rendered in front of the headline. `X<h1>First</h1>` should not work, though, because while `h1` is not directly a block tag, it still [may not occur in `p`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements), for example. See the fix provided in 4f89378. Let me know what you think and thanks for your effort!

Best, Alex
